src/components/trade/PriceImpact/utils.ts

```diff
@@ -0,0 +1,107 @@
+import BigNumber from 'bignumber.js'
+
+import { parseBigNumber } from 'utils'
+import { ONE_HUNDRED_BIG_NUMBER, ONE_BIG_NUMBER, ZERO_BIG_NUMBER } from 'const'
+
+import { PriceImpactArgs, PriceImpactArgsBase } from '../PriceImpact'
+
+// Limit as to where price colour changes
+const HIGH_PRICE_IMPACT_THRESHOLD = new BigNumber(5)
+const MID_PRICE_IMPACT_THRESHOLD = new BigNumber(3)
+const LOW_PRICE_IMPACT_THRESHOLD = ONE_BIG_NUMBER
+// TODO: remove for slippage toggle
+// 0.5% placeholder slippage
+const PLACEHOLDER_SLIPPAGE = new BigNumber(0.005)
+
+const determineImpactLevel = (impact: BigNumber | null): number => {
+  let impactLevel = 0
+  if (impact) {
+    if (impact.gte(HIGH_PRICE_IMPACT_THRESHOLD)) {
+      impactLevel = 3
+    } else if (impact.gte(MID_PRICE_IMPACT_THRESHOLD)) {
+      impactLevel = 2
+    } else if (impact.lt(LOW_PRICE_IMPACT_THRESHOLD)) {
+      impactLevel = 1
+    }
+  }
+
+  return impactLevel
+}
+
+// sets colour of strong tag presenting price impact
+const getImpactColourClass = (impact: BigNumber | null): string => {
+  const impactLevel = determineImpactLevel(impact)
+
+  let className = ''
+  if (impactLevel) {
+    if (impactLevel === 3) {
+      className = 'highImpact'
+    } else if (impactLevel === 2) {
+      className = 'midImpact'
+    } else if (impactLevel === 1) {
+      className = 'lowImpact'
+    }
+  }
+
+  return className
+}
+
+function determinePriceWarning(params: PriceImpactArgs, impact: BigNumber | null): string | null {
+  const { limitPrice, fillPrice, bestAskPrice } = params
+
+  const limitPriceBigNumber = limitPrice && parseBigNumber(limitPrice)
+  const fillPriceRounded = fillPrice?.toFixed(5)
+
+  // CASE 1 & 2: no warning
+  // round fillPrice to 5 decimals points and string compare against limit
+  if (
+    !limitPriceBigNumber ||
+    limitPriceBigNumber.isZero() ||
+    !fillPrice ||
+    !bestAskPrice ||
+    fillPriceRounded === limitPrice
+  )
+    return null
+
+  // Is user's limit price less than the suggested fill price?
+  const isLimitPriceLessThanFillPrice = limitPriceBigNumber.lt(fillPrice)
```

Comment:
Some declarative way to name this `isLimitPriceLessThanFillPrice` is `orderWontBeFullyExecuted`

Reply:
I will change this but imo this is totally subjective - I understand more clearly my variable name